### PR TITLE
Refactor Wars Episode IV: A New Hope

### PR DESCRIPTION
**Pull request by GawainLynch**
A long time ago in an RFC far, far away…

**Episode IV**

**A New Hope**

It is a period of refactor war.

Rebel developers, striking from a 
hidden base, have won their first 
victory against the evil 
Monolithic Code Empire.

During the battle, Rebel spies 
managed to steal secret plans to 
the Empire's ultimate weapon, 
Bolt v2.3, a heavily refactored 
release with enough power to 
build an entire web site.

Pursued by the Empire's sinister
agents, Princess Gawain races 
home aboard her starship, 
custodian of the stolen plans 
that can save her people 
and restore stability to the 
[codebase....](http://www.starwars.com/games-apps/star-wars-crawl-creator/?55d5e39ee4b0be1e479fdbf2)

star-wars-wallpaper-a-new-hope-8

### CODE REVIEW

File: src/AccessControl/Permissions.php [hunk 1]
```diff
@@ -515,7 +515,7 @@ public function isAllowed($what, $user, $content = null, $contentId = null)
     {
         if (is_array($content)) {
             $contenttypeSlug = $content['slug'];
-        } elseif ($content instanceof \Bolt\Content) {
+        } elseif ($content instanceof \Bolt\Legacy\Content) {
```
Comment (CarsonF) [inline src/AccessControl/Permissions.php]: Don't you need to check for the new class as well?

File: tests/phpunit/unit/Cache/CacheTest.php [hunk 2]
```diff
@@ -21,6 +21,7 @@ class CacheTest extends BoltUnitTest
 
     public function setUp()
     {
+$this->getApp();
```
Comment (CarsonF) [inline tests/phpunit/unit/Cache/CacheTest.php]: :)

File: src/AccessControl/Permissions.php (hunk 1 above)
Comment (GawainLynch) [inline src/AccessControl/Permissions.php]: Try it:
```php
class A{}
class B extends A{}
class C extends B {}

class_alias('A', 'AAA');

$a = new A();
$b = new B();
$c = new C();
$aaa = new AAA();

if ($a instanceof A) {
    echo "a is an instance of A\n";
}

if ($b instanceof A) {
    echo "b is an instance of A\n";
}

if ($c instanceof A) {
    echo "c is an instance of A\n";
}

if ($aaa instanceof A) {
    echo "aaa is an instance of A\n";
}

if ($aaa instanceof AAA) {
    echo "aaa is an instance of AAA\n";
}
```

File: tests/phpunit/unit/Cache/CacheTest.php (hunk 2 above)
Comment (GawainLynch) [inline tests/phpunit/unit/Cache/CacheTest.php]: Derp…

File: src/AccessControl/Permissions.php (hunk 1 above)
Comment (CarsonF) [inline src/AccessControl/Permissions.php]: I meant `Bolt\Storage\Entity\Content`...I guess we aren't using that yet though

Comment (GawainLynch) [inline src/AccessControl/Permissions.php]: Correct

Comment (GawainLynch) [inline src/AccessControl/Permissions.php]: This whole thing is just a first step in transitioning this stuff so it isn't stuck in someone's branch

**Comment by rarila** [comment]
As a member of the Javascript Order I was heavily hit by the Force while  imaging @GawainLynch with princess hair buns! – No, wait. The longer I thing about it… that explains everything…
See yourself: → :koala: ←

**Comment by Mikescops** [comment]
I don't understand the code but i approve the order #4000 from the :koala: mania !

**Comment by GawainLynch** [comment]
:+1:

**Pull request merged**